url-matchers.go

```diff
@@ -122,7 +123,22 @@ func unique[T comparable](items []T) []T {
 // and a function to actually do the matching and producing of the *URL
 type URLMatcher struct {
 	Type string
-	Fn   func(*Node) *URL
+	Fn   func(*Node) []*URL
+}
+
+func SingleURLMatcher(typ string, fn func(*Node) *URL) URLMatcher {
+	wrapperFn := func(n *Node) []*URL {
```

Comment:
I could also extract `wrapperFn` into public function `SingleMatcherFn` to allow constructs like `URLMatcher{"string", SingleMatcherFn(func(){...})}`, if you like.